pkg/addons/addons.go

```diff
@@ -0,0 +1,245 @@
+	b, err := strconv.ParseBool(val)
+	if err != nil {
+		return err
+	}
+	if m.Addons == nil {
+		m.Addons = map[string]bool{}
+	}
+	m.Addons[name] = b
+	return nil
+}
+
+// enableOrDisableAddon updates addon status executing any commands necessary
+func enableOrDisableAddon(name, val, profile string) error {
+	enable, err := strconv.ParseBool(val)
+	if err != nil {
+		return errors.Wrapf(err, "parsing bool: %s", name)
+	}
+	addon := assets.Addons[name]
+
+	// check addon status before enabling/disabling it
+	alreadySet, err := isAddonAlreadySet(addon, enable)
+	if err != nil {
+		out.ErrT(out.Conflict, "{{.error}}", out.V{"error": err})
+		return err
+	}
+	//if addon is already enabled or disabled, do nothing
+	if alreadySet {
+		return nil
+	}
+
+	if name == "istio" && enable {
+		minMem := 8192
+		minCpus := 4
+		memorySizeMB := pkgutil.CalculateSizeInMB(viper.GetString("memory"))
+		cpuCount := viper.GetInt("cpus")
+		if memorySizeMB < minMem || cpuCount < minCpus {
+			out.WarningT("Enable istio needs {{.minMem}} MB of memory and {{.minCpus}} CPUs.", out.V{"minMem": minMem, "minCpus": minCpus})
+		}
+	}
+
+	// TODO(r2d4): config package should not reference API, pull this out
+	api, err := machine.NewAPIClient()
+	if err != nil {
+		return errors.Wrap(err, "machine client")
+	}
+	defer api.Close()
+
+	//if minikube is not running, we return and simply update the value in the addon
+	//config and rewrite the file
+	if !cluster.IsMinikubeRunning(api) {
+		return nil
+	}
+
+	cfg, err := config.Load(profile)
+	if err != nil && !os.IsNotExist(err) {
+		exit.WithCodeT(exit.Data, "Unable to load config: {{.error}}", out.V{"error": err})
+	}
+
+	host, err := cluster.CheckIfHostExistsAndLoad(api, cfg.Name)
+	if err != nil {
+		return errors.Wrap(err, "getting host")
+	}
+
+	cmd, err := machine.CommandRunner(host)
+	if err != nil {
+		return errors.Wrap(err, "command runner")
+	}
+
+	data := assets.GenerateTemplateData(cfg.KubernetesConfig)
+	return enableOrDisableAddonInternal(addon, cmd, data, enable)
+}
+
+func isAddonAlreadySet(addon *assets.Addon, enable bool) (bool, error) {
+	addonStatus, err := addon.IsEnabled()
+
+	if err != nil {
+		return false, errors.Wrap(err, "get the addon status")
+	}
+
+	if addonStatus && enable {
+		return true, nil
+	} else if !addonStatus && !enable {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func enableOrDisableAddonInternal(addon *assets.Addon, cmd command.Runner, data interface{}, enable bool) error {
+	var err error
+
+	if enable {
+		for _, addon := range addon.Assets {
+			var addonFile assets.CopyableFile
+			if addon.IsTemplate() {
+				addonFile, err = addon.Evaluate(data)
+				if err != nil {
+					return errors.Wrapf(err, "evaluate bundled addon %s asset", addon.GetAssetName())
+				}
+
+			} else {
+				addonFile = addon
+			}
+			if err := cmd.Copy(addonFile); err != nil {
+				return errors.Wrapf(err, "enabling addon %s", addon.AssetName)
+			}
+		}
+	} else {
```

Comment:
Can this be refactored with less indentation for readability?

The easy win here would be to return if enabled, so that the disable state is dedented one level. 

Looking more closely, it seems like the loops are nearly identical, so you may be able to simply generate an array of addonFiles outside of the if statement, so that the only code that needs to be indented is `cmd.Copy` vs `cmd.Remove`

Reply:
done, there shouldn't be any duplication now